joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -214,9 +214,14 @@ class JointStatePublisher():
                 elif name in self.dependent_joints:
                     param = self.dependent_joints[name]
                     parent = param['parent']
-                    joint = self.free_joints[parent]
                     factor = param.get('factor', 1)
                     offset = param.get('offset', 0)
+                    while parent in self.dependent_joints:
+                        param = self.dependent_joints[parent]
+                        parent = param['parent']
+                        factor *= param.get('factor', 1)
+                        # the offset is relative only to the first parent
```

Comment:
I don't even use offset in my application, so it is definitely not a problem to me.

I just wanted to make a useful change for the community and I thank you for your time and experience in this field.